### ISSUE TITLE
Valid json rasies error when triggering with config in 2.1.1

### ISSUE THREAD

**Issue by Jaxing** [apache/airflow#16908]
**Apache Airflow version**:

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): 1.18+

**Environment**:

- **Cloud provider or hardware configuration**: GCP
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
Tried to trigger a dag with config but got an error saying "Invalid JSON configuration, must be a dict"
This worked in 2.1.0
![image](https://user-images.githubusercontent.com/8545525/125070083-eaed7680-e0b7-11eb-954b-c3e149a98bd8.png)

**What you expected to happen**:

**How to reproduce it**:
Running airflow 2.1.1 try trigger a dag with config

**Anything else we need to know**:
How often does this problem occur? Once? Every time etc?
Always
Any relevant logs to include? Put them here in side a detail tag:
<details><summary>webserver_logs</summary>
```
10.60.7.19 - - [09/Jul/2021:10:00:46 +0000] "POST /trigger?dag_id={dag_id}&origin=%2Ftree%3Fdag_id%3D{dag_id}HTTP/1.1" 200 19096 "https://{airflow-domain}/trigger?dag_id={dag_id}&origin=%2Ftree%3Fdag_id%3D{dag_id}" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36"
10.60.7.19 - - [09/Jul/2021:10:00:46 +0000] "GET /static/dist/lint.css HTTP/1.1" 304 0 "https://{airflow-domain}/trigger?dag_id={dag_id}&origin=%2Ftree%3Fdag_id%3D{dag_id}" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36"
10.60.7.19 - - [09/Jul/2021:10:00:46 +0000] "GET /static/dist/codemirror.css HTTP/1.1" 304 0 "https://{airflow-domain}/trigger?dag_id={dag_id}&origin=%2Ftree%3Fdag_id%3D{dag_id}" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36"
```
</details>

**Comment by potiuk** [comment]
Yep this issue is already reported and is fixed in upcoming 2.1.2 (next few days)

**Comment by potiuk** [comment]
Duplicate of #16810 to be precise.
